Examples/Io/Csv/include/ActsExamples/Io/Csv/CsvSeedWriter.hpp

```diff
@@ -0,0 +1,107 @@
+#include "ActsExamples/Framework/WriterT.hpp"
+#include "ActsFatras/EventData/Barcode.hpp"
+
+#include <fstream>
+
+using namespace Acts::UnitLiterals;
+
+namespace ActsExamples {
+
+/// @class CsvSeedWriter
+///
+/// Write out the seed reconstructed by the seeding algorithm in
+/// comma-separated-value format.
+///
+/// This writes one file per event into the configured output directory. By
+/// default it writes to the current working directory.
+/// Files are named using the following schema
+///
+///     event000000001-seed.csv
+///     event000000002-seed.csv
+///
+/// and each line in the file corresponds to one seed.
+class CsvSeedWriter : public WriterT<TrackParametersContainer> {
+ public:
+  using HitParticlesMap = IndexMultimap<ActsFatras::Barcode>;
+  using HitSimHitsMap = IndexMultimap<Index>;
+
+  struct Config {
+    /// Input estimated track parameters collection.
+    std::string inputTrackParameters;
+    /// Input seed collection.
+    std::string inputSimSeeds;
+    /// Input collection of simulated hits.
+    std::string inputSimHits;
+    /// Input hit-particles map collection.
+    std::string inputMeasurementParticlesMap;
+    /// Input collection to map measured hits to simulated hits.
+    std::string inputMeasurementSimHitsMap;
+    /// output filename.
+    std::string fileName = "Seed.csv";
+    /// output directory
+    std::string outputDir;
+  };
+
+  /// Constructor
+  ///
+  /// @param config Configuration struct
+  /// @param level Message level declaration
+  CsvSeedWriter(const Config& config,
+                Acts::Logging::Level level = Acts::Logging::INFO);
+
+  /// Get readonly access to the config parameters
+  const Config& config() const { return m_cfg; }
+
+ protected:
+  /// @brief Write method called by the base class
+  /// @param [in] ctx is the algorithm context for event information
+  /// @param [in] trackParams are parameters to write
+  ProcessCode writeT(const AlgorithmContext& ctx,
+                     const TrackParametersContainer& trackParams) override;
+
+ private:
+  Config m_cfg;  ///< The config class
+
+  ReadDataHandle<SimParticleContainer> m_inputParticles{this, "InputParticles"};
+  ReadDataHandle<SimSeedContainer> m_inputSimSeeds{this, "InputSimSeeds"};
+  ReadDataHandle<SimHitContainer> m_inputSimHits{this, "InputSimHits"};
+  ReadDataHandle<HitParticlesMap> m_inputMeasurementParticlesMap{
+      this, "InputMeasurementParticlesMap"};
+  ReadDataHandle<HitSimHitsMap> m_inputMeasurementSimHitsMap{
+      this, "InputMeasurementSimHitsMap"};
+
+  /// @brief Struct for brief seed summary info
+  ///
+  struct seedInfo {
+    std::size_t seedId = 0;
+    ActsFatras::Barcode particleId;
+    float seedPt = -1;
+    float seedPhi = 0;
+    float seedEta = 0;
+    float vertexZ = 0;
+    float quality = -1;
+    std::vector<Acts::Vector3> globalPosition;
+    float truthDistance = -1;
+    std::string seedType = "unknown";
+    ProtoTrack measurementsID;
```

Comment:
could this rather point to a proto track csv?

Reply:
I can explicitly replace it by a `boost::container::small_vector<Index, 3>;` if you prefer ? I am not sure what you mean here

Reply:
The ProtoTrack contains basically only contains 3 measurement ID (the 3 hits of the seed). From my experience having worked with those CSV for the past few month having those in the same file make it much more human readable (it let you easily know if seed share hits). 

Internally the prototrack are just boost::container::small_vector<Index, 3> so not converting them would be the easiest from my side (and that let us change the seed hit size in the futur).